src/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -19,7 +22,23 @@ export function PrimaryLayout ({
   showSideMenu = true,
   showAlphaFlag = true
 }) {
-  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery)
+  const { setIsConnected, isConnected } = useConnectionContext()
+
+  const onError = ({ graphQLErrors }) => {
+    const hposConnection = graphQLErrors.hposConnection
+    setIsConnected(hposConnection)
+  }
+
+  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery, { pollInterval: 30000, onError })
+  const { newMessage } = useFlashMessageContext()
+
+  useEffect(() => {
+    if (!isConnected) {
+      newMessage('Your Holoport is currently unreachable.', 0)
+    } else {
+      newMessage('', 0)
+    }
+  }, [isConnected, setIsConnected, newMessage])
```

Comment:
It's so nice we can write this now and don't have to add a whole extra layer of `hasMessageBeenDisplayed` logic :)